tasks/kernels.py

```diff
@@ -84,11 +94,11 @@ def propagate_error(wasm_results, native_results, num_proc):
     ]
     native_times = [
         native_results[kern][0]["ActualTime"].tolist()[num_proc]
-        for kern in native_results.keys()
+        for kern in wasm_results.keys()
```

Comment:
We check beforehand that `native_results` and `wasm_results` have the same keys.

We switch to using the same dictionary in the loop to guarantee we iterate keys in the same order.